serde/src/private/de.rs

```diff
@@ -2587,6 +2587,28 @@ where
     }
 }
 
+impl<'de, E> IdentifierDeserializer<'de, E> for i64
+where
+    E: Error,
+{
+    type Deserializer = <i64 as IntoDeserializer<'de, E>>::Deserializer;
+
+    fn from(self) -> Self::Deserializer {
+        self.into_deserializer()
+    }
+}
+
+impl<'de, E> IdentifierDeserializer<'de, E> for bool
```

Comment:
If I understand correctly (based on what test fails if I delete this impl), this is to support the following?

```rust
#[derive(Deserialize)]
#[serde(field_identifier)]
enum Field {
    Aaa,
    Bbb,
    Other(bool),
}
```

Is that a necessary part of this PR? I'd prefer if this PR had no impact on `field_identifier` generated code, since integer/boolean continue to not be supported as a field name, and do not have a compelling justification as enum tags do.

In general I would like this PR to avoid introducing support for anything other than what is required for `rename = 1` and `rename = true` on adjacently tagged enum variants and internally tagged enum variants. That includes no `serde(variant_identifier)` changes for now; those can be independently motivated if someone would like.